pulp_container/app/tasks/sync_stages.py

```diff
@@ -349,6 +379,117 @@ def create_blob(self, blob_data, deferred_download=True):
 
         return blob_dc
 
+    async def create_signatures(self, man_dc, signature_source):
+        """
+        Create signature declarative artifacts.
+
+        Signatures are currently supported only for image manifests and not manifest lists.
+
+        For sigstore, signatures are downloaded from a specified url. The number of signatures is
+        unknown upfront, need to download them in order by incrementing the index until hit 404.
+
+        Args:
+            man_dc: Declarative content instance of a related Manifest
+            signature_source: Source where to get signatures from
+        """
+
+        def _extract_data_from_signature(signature_raw):
+            gpg = gnupg.GPG()
+            crypt_obj = gpg.decrypt(signature_raw)
+            if not crypt_obj.data:
+                log.info(
+                    _(
+                        "It is not possible to read the signed document, GPG error: {}".format(
+                            crypt_obj.stderr
+                        )
+                    )
+                )
+                return
+
+            try:
+                sig_json = json.loads(crypt_obj.data)
+            except Exception as exc:
+                log.info(
+                    _(
+                        "Signed document cannot be parsed to create a signature for {}."
+                        " Error: {}".format(man_dc.content.digest, str(exc))
+                    )
+                )
+                return
+
+            # Currently, the only supported type is ATOMIC.
+            if sig_json["critical"]["type"] != SIGNATURE_TYPE.ATOMIC_FULL:
+                log.info(
+                    _(
+                        "Unsupported signature type {}, signature is not synced for {}.".format(
+                            signature_json["critical"]["type"], man_dc.content.digest
+                        )
+                    )
+                )
+                return
+
+            sig_json["signing_key_id"] = crypt_obj.key_id
+            sig_json["signature_timestamp"] = crypt_obj.timestamp
+            return sig_json
+
+        signature_dcs = []
+        if signature_source == SIGNATURE_SOURCE.SIGSTORE:
+            man_digest_reformatted = man_dc.content.digest.replace(":", "=")
+            # should the namespaced_upstream_name be used to grab a signature?
```

Comment:
This was done to follow dockerhub's `library` conventions. We can leave this as you have since docker has docker content trust for signing, which we do not support https://docs.docker.com/engine/security/trust/